independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BuiltinBean.java

```diff
@@ -170,13 +170,28 @@ interface Generator {
 
     }
 
+    public static class ValidatorContext {
+        final BeanDeployment beanDeployment;
+        final InjectionTargetInfo injectionTarget;
+        final InjectionPointInfo injectionPoint;
+        final Consumer<Throwable> errors;
+
+        public ValidatorContext(BeanDeployment beanDeployment, InjectionTargetInfo injectionTarget,
```

Comment:
I know I'm annoying but
```java
public record ValidatorContext(InjectionTargetInfo injectionTarget, InjectionPointInfo injectionPoint, Consumer<Throwable> errors) {}
```
is tempting ;-).

NOTE: You don't need the `BeanDeployment` - see my other comment.

Reply:
Good point. I just blindly copied the `GeneratorContext` :-) I'll change both.

Reply:
Turned both `GeneratorContext` and `ValidatorContext` into records.